### ISSUE TITLE
The timezone mess...

### ISSUE THREAD

**Issue by kamil4** [LycheeOrg/Lychee-Laravel#132]
I just faced an extremely annoying problem related to... an upcoming daylight saving time change (tonight in the US). All of a sudden, Laravel would refuse to add entries to the logs table, generating the following error messages in the storage log:

`[2019-03-10 02:36:17] local.ERROR: SQLSTATE[22007]: Invalid datetime format: 1292 Incorrect datetime value: '2019-03-10 02:36:17' for column 'updated_at' at row 1`

Since I couldn't figure it out, I decided to reinitialize the database, especially since earlier today I had a disk full problem on my server so I thought maybe it got corrupted. Anyway, that did not fix it. I started searching around and I found the following entry on stackoverflow:

https://stackoverflow.com/questions/37203899/sqlstate22007-invalid-datetime-format-1292-incorrect-datetime-value-2008-0

It turns out it has to do with invalid timestamps! 2am on Sunday, March 10 2019 is an invalid time for my time zone (because the local time jumps forward from 2am to 3am due do daylight saving time change). Fair enough, but I wasn't doing my development at 2am but at 8pm on Saturday evening. Somewhere in Lychee or in Laravel, there's a confusion between local time and UTC. Local time (8pm) gets coverted to UTC (2am), but then gets reinterpreted again as local time, which is invalid for one hour a year.

I don't know if this is worth tracking down and fixing, for a problem that only manifests itself an hour a year, but let me tell you, when you are unlucky enough to encounter it, it is extremely annoying...

**Comment by d7415** [comment]
At a glance it doesn't look like Lychee is doing anything with timezones, other than the setting in `config/app.php`. Your link suggests that TIMESTAMP is stored as UTC, so that explains one conversion - I wonder if Laravel is doing some conversion and then forgetting.

Or, perhaps, it's a confusion between the system timezone, the timezone in app.php and UTC in the DB.

Do your system and app.php timezones match?

**Comment by kamil4** [comment]
I didn't know about `config/app.php`. I should experiment with it, although now that the "magic" hour has passed, I'll need to rig something to try to insert those "weird" dates. For what it's worth, I searched a little more today and I found two more extremely relevant questions on stackoverflow:

https://stackoverflow.com/questions/55084109/laravel-sqlstate22007-invalid-datetime-format-1292-incorrect-datetime-value
https://stackoverflow.com/questions/55083933/mysql-weird-date-error-incorrect-datetime-value

Both were asked... yesterday! So I clearly wasn't the only person affected... There is a follow-up in the first link that suggests that Laravel apps should force timezone to UTC when interacting with MySQL to avoid such issues.
